apps/src/JSInterpreter.js

```diff
@@ -40,16 +42,45 @@ var JSInterpreter = module.exports = function (options) {
     options.code += '\nwhile (true) { var obj = getCallback(); ' +
       'if (obj) { var ret = obj.fn.apply(null, obj.arguments ? obj.arguments : null);' +
                  'setCallbackRetVal(ret); }}';
+
+    codegen.createNativeFunctionFromInterpreterFunction = function (intFunc) {
+      return function () {
+        if (self.initialized()) {
+          self.queueEvent(intFunc, arguments);
+          
+          if (self.executeLoopDepth === 0) {
+            // Execute the interpreter and if a return value is sent back from the
+            // interpreter's event handler, pass that back in the native world
+
+            // NOTE: the interpreter will not execute forever, if the event handler
+            // takes too long, executeInterpreter() will return and the native side
+            // will just see 'undefined' as the return value. The rest of the interpreter
+            // event handler will run in the next onTick(), but the return value will
+            // no longer have any effect.
+            self.executeInterpreter(false, true);
+            return self.lastCallbackRetVal;
+          }
+        }
+      };
+    };
   }
 
-  var self = this;
   var initFunc = function (interpreter, scope) {
+    // Store Interpreter on JSInterpreter
+    self.interpreter = interpreter;
+    // Store globalScope on JSInterpreter
     self.globalScope = scope;
+    // Override Interpreter's get/set Property functions with JSInterpreter
+    self.baseGetProperty = interpreter.getProperty;
+    interpreter.getProperty = _.bind(self.getProperty, self);
+    self.baseSetProperty = interpreter.setProperty;
+    interpreter.setProperty = _.bind(self.setProperty, self);
```

Comment:
No, that's a good point. We have quite a bit of `_.bind` in our code base, but as long as we're comfortable with browser support for `Function.prototype.bind` (which we must be, because we a lot more use of that method), we should probably eliminate `_.bind`. I'll start removing it from this code in my next PR.